### PR TITLE
Use the connect-the-dots algorithm by default in DetermineBonds

### PR DESCRIPTION
**Pull request by greglandrum**
This was a suggestion by @rogersayle at the 2023 RDKit UGM: use the connect the dots algorithm instead of the xyz2mol VdW method by default in DetermineBonds

### CODE REVIEW

File: Code/GraphMol/DetermineBonds/DetermineBonds.cpp [hunk 1]
```diff
@@ -170,8 +171,10 @@ void determineConnectivity(RWMol &mol, bool useHueckel, int charge,
   }
   if (useHueckel) {
     connectivityHueckel(mol, charge);
-  } else {
+  } else if (useVdw) {
     connectivityVdW(mol, covFactor);
+  } else {
+    ConnectTheDots(&mol, ctdIGNORE_H_H_CONTACTS);
```
Comment (richardjgowers) [inline Code/GraphMol/DetermineBonds/DetermineBonds.cpp:177]: As a hackathon idea, would it be worth extending ConnectTheDots to take optional vdw radii so it could also be the useVdw code path?

Comment (greglandrum) [reply Code/GraphMol/DetermineBonds/DetermineBonds.cpp:177]: I'm not sure it's worth it. As Roger pointed out: figuring out whether or not two atoms should be connected ought not to be that tricky (at least as long as we stick to organic molecules)

**Comment by rogersayle** [comment]
One minor issue (distraction) that should be cleared up is the distinction between "Van der Waals" and "covalent radii",  Van der Waals (VdW) is the lowest energy separation of two non-bonded atoms (i.e. contact distance between ligand and protein), where as covalent radii are the related to the distance that below which to atoms should be considered bonded(i.e. exhibit significant electron sharing).  Hence VdW is a minimum distance that is typically larger than covalent radius which is a maximum distance.  Two atoms seprated by a distance between these two values are considered to be bumping (energetically disfavored).

TLDR; For "connecting the dots" it's covalent radii that are relevant.  (useVdw and connectivityVdW are misnamed).

**Review by ptosco — COMMENTED** [review]
The multiple `bool` parameters `useVdw` and `useHueckel` are not very pretty, particularly as it is not obvious what will happen if both are `true`.
How about leaving the current signatures as they are and make them deprecated, and add another overload that takes a `connectivityMethod` `enum`, e.g.:
```
determineConnectivity(RWMol &mol, double covFactor, int charge, unsigned int connectivityMethod)
```

**Comment by greglandrum** [comment]
> The multiple `bool` parameters `useVdw` and `useHueckel` are not very pretty, particularly as it is not obvious what will happen if both are `true`. How about leaving the current signatures as they are and make them deprecated, and add another overload that takes a `connectivityMethod` `enum`, e.g.:
> 
> ```
> determineConnectivity(RWMol &mol, double covFactor, int charge, unsigned int connectivityMethod)
> ```

I agree with you and thought about that, but I would like to deprecate the useVdw method entirely in favor of connect-the-dots and I think the enum makes useVdw look more permanent

**Review by ptosco — APPROVED** [review]
@greglandrum OK, I see. If I read the comment in `DetermineBonds.h` correctly, if `useHueckel` is `true` any other `bool` potentially set to `true` is ignored, so this is already documented after all.

**Comment by rogersayle** [comment]
It's hopefully easy (for you guys) to test from python, but do any of the test cases in GraphMol/DetermineBonds/test_data (or QM9) actually produce different results for useHueckel vs useVdW vs ConnectTheDots?  I'm working my way through the list visualizing them with RasMol, and haven't seen anything challenging/incorrect so far.

**Comment by greglandrum** [comment]
#hackathon
